src/client/services/search.ts

```diff
@@ -0,0 +1,71 @@
+import { objectType } from "../utils/types";
+import baseService from "./base";
+
+const findByTitle = async (searchString: string) => {
+	console.log(`Search string`, searchString);
+	try {
+		const response = await baseService.get(`/api/search/title/${searchString}`);
+		console.log(`response`, response);
+		if (response.result.length === 0) {
+			return response.status(204).json({ message: "No patterns found" });
+		}
+		return response.result;
+	} catch (error) {
+		console.error("Error fetching patterns:", error);
+		throw error;
+	}
+};
+
+const findByTags = async (payload: objectType[]) => {
+	try {
+		const response = await baseService.post(`/api/search/tag`, {
+			tagList: JSON.stringify(payload),
+		});
+		console.log(`response`, response);
+		if (response.finalPatterns.length === 0) {
+			return response.status(204).json({ message: "No patterns found" });
+		}
+		return response;
+	} catch (error) {
+		throw error;
+	}
+};
+
+const addNewPattern = async (payload: {
+	title: string;
+	content: string;
+	author_id: string;
+}) => {
+	try {
+		console.log(`Adding pattern....`);
+		const response = await baseService.post("/api/patterns", payload);
```

Comment:
I'm going to disagree with this because it would muddy the calls by having to do a patternsBaseUri + {id} and even thought it's repetitive, I like the clarity it gives when I read the code. It also helps me navigate between my files, as the path is my file structure so I can find where my calls are going to much faster than if I have to scroll to the top to find where that variable is taking me. Thoughts?